editoast/src/generated_data/error/mod.rs

```diff
@@ -246,30 +268,117 @@ fn get_insert_errors_query(obj_type: ObjectType) -> &'static str {
     }
 }
 
-/// Insert a heterogeneous list of infra errors in DB with a minimum number of queries
-async fn insert_errors(
-    conn: &mut PgConnection,
-    infra_id: i64,
-    errors: Vec<InfraError>,
-) -> Result<()> {
-    let mut errors_by_type: HashMap<_, Vec<_>> = Default::default();
+/// This function dispatch errors to the right layer
+fn dispatch_errors_to_layers(
+    errors: Vec<ErrorWithHash>,
+) -> HashMap<ObjectType, Vec<ErrorWithHash>> {
+    let mut errors_by_layer: HashMap<_, Vec<_>> = Default::default();
     for error in errors {
-        errors_by_type
-            .entry(error.get_type())
+        errors_by_layer
+            .entry(error.error.get_type())
             .or_default()
-            .push(to_value(error).unwrap());
+            .push(error);
     }
+    errors_by_layer
+}
+
+/// Retrieve the current errors hash for a given infra
+async fn retrieve_current_errors_hash(
+    conn: &mut PgConnection,
+    infra_id: i64,
+) -> Result<Vec<String>> {
+    use crate::tables::infra_layer_error::dsl;
+    Ok(dsl::infra_layer_error
+        .filter(dsl::infra_id.eq(infra_id))
+        .select(dsl::info_hash)
+        .load(conn)
+        .await?)
+}
+
+/// Remove a list of errors given an infra and a list of error hashes
+async fn remove_errors_from_hashes(
+    conn: &mut PgConnection,
+    infra_id: i64,
+    errors_hash: &Vec<&String>,
+) -> Result<()> {
+    use crate::tables::infra_layer_error::dsl;
+    let nb_deleted = diesel::delete(
+        dsl::infra_layer_error
+            .filter(dsl::infra_id.eq(infra_id))
+            .filter(dsl::info_hash.eq_any(errors_hash)),
+    )
+    .execute(conn)
+    .await?;
+    assert_eq!(nb_deleted, errors_hash.len());
```

Comment:
Is it ok to panic in production? You may prefer `debug_assert_eq!()` instead (see [documentation](https://doc.rust-lang.org/std/macro.debug_assert_eq.html))?